packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -602,27 +585,43 @@ function ensureRootIsScheduled(root: FiberRoot) {
   root.callbackNode = callbackNode;
 }
 
-function runRootCallback(root, callback, isSync) {
-  const prevCallbackNode = root.callbackNode;
-  let continuation = null;
-  try {
-    continuation = callback(isSync);
-    if (continuation !== null) {
-      return runRootCallback.bind(null, root, continuation);
-    } else {
-      return null;
+// This is the entry point for every concurrent task.
+function performWorkOnRoot(root, isSync) {
+  // Determine the next expiration time to work on, using the fields stored
+  // on the root.
+  let expirationTime = getNextRootExpirationTimeToWorkOn(root);
+  if (expirationTime !== NoWork) {
+    if (expirationTime !== Sync) {
+      // Since we know we're in a React event, we can clear the current
+      // event time. The next update will compute a new event time.
+      currentEventTime = NoWork;
```

Comment:
Is this a React event?

(is any queued async update considered an "event"?)

Reply:
The "event" here refers to a scheduled callback with the scheduler. I.e. clean stack.